### PR TITLE
feat: adding docker file for building image

### PR DESCRIPTION
**Pull request by CDaRip2U**
Signed-off-by: Chris Replogle <Chris.Replogle@sas.com>

### CODE REVIEW

**Comment by CDaRip2U** [comment]
This is just the Dockerfile and the .dockerignore to build a image from the odpi/egeria image and include JUST the application connector jar.  Excluded: javadoc.jar, sources.jar.

File: .dockerignore
Comment (planetf1) [inline .dockerignore:1]: We can add the SPDX header

File: Dockerfile
Comment (planetf1) [inline Dockerfile:8]: We will want to retrieve the same version of egeria as that we are building.

We should make this version an argument - docker now supports this - see https://www.jeffgeerling.com/blog/2017/use-arg-dockerfile-dynamic-image-specification . Default could be 'latest' in case someone runs it without arg.

File: Dockerfile
Comment (planetf1) [inline Dockerfile:13]: see above - just need this earlier before the from

File: Dockerfile
Comment (planetf1) [inline Dockerfile:14]: Like with react ui, we don't use these yet. They were setup in the base where the maven process pulls them out. I'm hoping at some point we can do this in the build yaml instead as extra metadata could be useful

File: Dockerfile
Comment (planetf1) [inline Dockerfile:35]: probably the default port is not 8091.. but egeria?

File: Dockerfile
Comment (planetf1) [inline Dockerfile:33]: wrong url - shouldn't it point to this repo?

File: Dockerfile
Comment (planetf1) [inline Dockerfile:26]: not sure what's best for this metadata - main egeria project, or this repo

**Review by planetf1 — CHANGES_REQUESTED** [review]
Looks good ... to actually run the build we'll need a change to release.yaml or the merge build too (or I can do)

File: Dockerfile
Comment (CDaRip2U) [reply Dockerfile:35]: Agreed. expose 8080? and 8443?

File: Dockerfile
Comment (CDaRip2U) [reply Dockerfile:33]: That url is to this repo isn't it?

File: Dockerfile
Comment (CDaRip2U) [reply Dockerfile:35]: going to remove the port here and allow that to be up to the consumer for now.

File: .dockerignore
Comment (CDaRip2U) [reply .dockerignore:4]: Adding.

File: Dockerfile
Comment (planetf1) [inline Dockerfile:8]: How about having EGERIA_BASE_IMAGE=odpi/egeria -- so that a user override with their own repo if needed, but removing the latest tag as this overlaps effectively with 'version' above? (and is probably useful to keep seperate?)

File: Dockerfile
Comment (planetf1) [inline Dockerfile:14]: maybe 
FROM ${EGERIA_BASE_IMAGE}:${version} 

What you have works, but makes 'version' superfluous?

When we get around to extracting version from the code, having it seperate to the repo name probably is helpful.

File: .github/workflows/merge.yml
Comment (planetf1) [inline .github/workflows/merge.yml:20]: I'm thinking at some point we might be able to extract it from the pom (like we do with the egeria-react-ui build for npm) so it's in a single place, but this is fine for now.

File: .github/workflows/merge.yml
Comment (planetf1) [inline .github/workflows/merge.yml:71]: I wonder if we should add the environment variable here ideally to set the version (though the defaults match for now - it's probably better?) - especially since we already have a version in the script.

See https://github.com/docker/build-push-action/tree/releases/v1#build_args I think
build_args: version=${{ env.RELEASE_VERSION }}

File: .github/workflows/merge.yml
Comment (planetf1) [inline .github/workflows/merge.yml:81]: and add the version here too?

**Review by planetf1 — COMMENTED** [review]
I think this will work. I've added some comments I think might improve it slightly, but can merge with or without.

It's tricky to fully test until merged...

File: .github/workflows/merge.yml
Comment (CDaRip2U) [reply .github/workflows/merge.yml:71]: This ended up in the with section below as requested.

File: .github/workflows/merge.yml
Comment (CDaRip2U) [reply .github/workflows/merge.yml:82]: done, see below

File: Dockerfile
Comment (CDaRip2U) [reply Dockerfile:7]: Should this default to "latest" and allow the specific version to come in from the merge.yaml "env.RELEASE_VERSION"?

Comment (planetf1) [reply Dockerfile:7]: Yes I think that would be reasonable

**Review by planetf1 — APPROVED** [review]
Agree with your comment on default=latest.

I think it's fine to merge when you're happy & we can review the actions result -> https://github.com/odpi/egeria-connector-sas-viya/actions/workflows/merge.yml

**Pull request merged**
